pyomo/contrib/mindtpy/algorithm_base_class.py

```diff
@@ -805,6 +810,9 @@ def MindtPy_initialization(self):
             self.integer_list.append(self.curr_int_sol)
             fixed_nlp, fixed_nlp_result = self.solve_subproblem()
             self.handle_nlp_subproblem_tc(fixed_nlp, fixed_nlp_result)
+            self.int_sol_2_cuts_ind[self.curr_int_sol] = list(
+                range(1, len(self.mip.MindtPy_utils.cuts.oa_cuts) + 1)
+            )
```

Comment:
Why not just store the number of OA cuts for each of these solutions? It seems inefficient to store a list of indices when you can always construct a generator from one number, but maybe I'm wrong?

Reply:
`int_sol_2_cuts_ind` is a `dict`, with `self.curr_int_sol` list as the `key` and `index list` as the `value`. 
Do you want me to improve the definition of the key or the value? I think improving the value is doable.

Reply:
That's a good point. However, one integer is not enough. We need two integers denoting the starting index and ending index respectively. I will try to implement it.